### PR TITLE
L3Outside Multi External Network Support

### PR DESCRIPTION
**Pull request by christides11**
Support for alternative naming scheme for epgs.

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -29,3 +29,4 @@ testtools>=1.4.0
 mock
 tabulate>=0.7.5
 voluptuous!=0.13.0
+SQLAlchemy==1.4.47
```
Comment (tbachman) [inline test-requirements.txt:32]: Since we use AIM in different distros (Ubuntu 18.04, 20.04, RHEL7, RHEL8), I don't think we should specify the constraint this way. Instead, we may want to limit it to be below a certain version. That may take experimentation. We also should probably do that as a separate patch.

File: aim/aim_lib/nat_strategy.py [hunk 2]
```diff
@@ -624,23 +633,25 @@ def _select_domains(self, objs, vmm_domains=None, phys_domains=None):
                 if phys_domains is not None:
                     obj.physical_domains = phys_domains
 
-    def _create_nat_epg(self, ctx, l3out, vmm_domains=None, phys_domains=None):
-        objs = self._get_nat_objects(ctx, l3out)
+    def _create_nat_epg(self, ctx, l3out, vmm_domains=None, phys_domains=None,
+                        epg_name=None):
+        objs = self._get_nat_objects(ctx, l3out, epg_name)
```
Comment (tbachman) [inline aim/aim_lib/nat_strategy.py:638]: This is a nit, but you probably should pass this as a named kwarg: epg_name=epg_name. That would guard against any changes in the future that might introduce more positional arguments.

**Review by tbachman — CHANGES_REQUESTED** [review]
Pretty much LGTM, but a few nits. We probably want to break out the test-requirements.txt into a separate patch, and then just a NIT in one of the calls.

**Review by sayalinaval — APPROVED** [review]

File: test-requirements.txt [hunk 3]
```diff
@@ -28,4 +28,4 @@ testscenarios>=0.4
 testtools>=1.4.0
 mock
 tabulate>=0.7.5
-voluptuous!=0.13.0
+voluptuous!=0.13.0
```
Comment (tbachman) [inline test-requirements.txt:31]: Is there a change here? If not, you should be able to remove any changes to this file.

**Review by tbachman — APPROVED** [review]
LGTM!

**Review by sayalinaval — APPROVED** [review]

**Review by tbachman — APPROVED** [review]
